### ISSUE TITLE
Revisit how nodeType color mapping is auto-generated

### ISSUE THREAD

**Issue by aaronsarnat** [allenai/hierplane#3]
Currently, automatic `nodeType` styling (e.g. `color1`, `color2`, etc.) is incremented by the order that distinct `nodeType` values are found in the data. One drawback to this approach is that root and first-level children will not always be the same colors. This can be somewhat disorienting when visualizing multiple structures.

**Comment by codeviking** [comment]
>  One drawback to this approach is that root and first-level children will not always be the same colors.

Why would they be the same color? Doesn't that depend on their type?  If their type is the same, they'll be the same color, which is per spec / desired to my knowledge.  Am I missing something?

**Comment by aaronsarnat** [comment]
No, sorry. What I meant was from tree to tree. By default, I think it would be better if the root nodetype were always color1, its immediate children nodetypes color2, color3, etc. Unless specified manually.

I think hopping from tree to tree would be less visually jarring if there were a consistent default coloring of unique nodetypes ordered by their level in the hierarchy.

Not a top priority, but something I noticed as I was looking at different trees.

**Comment by codeviking** [comment]
> I think hopping from tree to tree would be less visually jarring if there were a consistent default coloring of unique nodetypes ordered by their level in the hierarchy.

We can only make that happen if we know what the node types are (i.e. someone has to provide the `nodeTypeToColor` mapping).  By definition an auto generated mechanism works from the node types it discerns from the current tree -- it has no understanding or awareness of the other trees that may or may not share that node type.

That said, if we support the ability for the user to cycle through multiple trees, we could generate an enforce a single map across trees that would resolve this for the immediate render.

In other words, I think the right way to solve this is to document `nodeTypeToStyle` so that users can specify this mapping and enforce it across renders.  I don't think the autogenerated mechanism can nor should enforce some sort of consistent node type to color mapping.

Does that make sense?

**Comment by aaronsarnat** [comment]
Yes, that makes sense, and I agree about documenting `nodeTypeToStyle`, but my suggestion wasn't about knowing anything about other trees besides the current one.

My suggestion was that we take hierarchy into account when incrementing the auto-assigned color style when it finds a new unique `nodeType` in the tree. Right now it seems that perhaps it steps through the tree and styles new node types from leaf up? I'd suggest we go from root down.

Here's a diagram that shows what I'm envisioning. Note how every root node has color1 style. And when it goes through its descendants, the first new node type it finds gets color2, then color3, etc.

![image](https://user-images.githubusercontent.com/8367927/33143614-4c59169e-cf6f-11e7-948c-036231cf8ac3.png)

**Comment by codeviking** [comment]
Ah! Great.  Yup, it's going leaf -> root, but that's extremely easy to fix.  We can mess around w/ that today during our polish jam session.
